test/e2e/config/e2e_conf.yaml

```diff
@@ -108,10 +108,6 @@ providers:
 variables:
   CNI: "/tmp/calico.yaml"
   KUBERNETES_VERSION: "v1.27.1"
-  # INIT_WITH_KUBERNETES_VERSION will be used here
-  # https://github.com/kubernetes-sigs/cluster-api/blob/bb377163f141d69b7a61479756ee96891f6670bd/test/e2e/clusterctl_upgrade.go#L170
-  # INIT_WITH_KUBERNETES_VERSION  used by management cluster upgrade and it is fixed to 1.23.8 because v1a4 capi does not support k8s version higher
-  INIT_WITH_KUBERNETES_VERSION: "v1.23.8"
```

Comment:
You cannot remove this if you want to use the variable. Just put a default here and then override with environment variables.